src/Util/Common.php

```diff
@@ -52,15 +52,57 @@ public static function isPharFile($path)
 
 
     /**
+     * Get absolute path, do not check exsit
+     *
+     * @param string $path The path to use.
+     *
+     * @return mixed
+     */
+    public static function absolutePath($path)
+    {
+        // Don't prefix absolute paths.
+        if (substr($path, 0, 1) !== '/') {
+            $path = getcwd().'/'.$path;
+        }
+
+        // Replace backslashes with forwardslashes.
+        $path = str_replace('\\', '/', $path);
+        // Combine multiple slashes into a single slash.
+        $path = preg_replace('/\/+/', '/', $path);
+
+        $parts = [];
+
+        foreach (explode('/', $path) as $part) {
+            // Ignore parts that have no value.
+            if (empty($part) === true || $part === '.') continue;
+            if ($part !== '..') {
+                // Cool, we found a new part.
+                array_push($parts, $part);
+            } else if (count($parts) > 0) {
+                // Going back up.
+                array_pop($parts);
+            } else {
+                // Path is outside of the root.
+                return false;
+            }
+        }
+
+        return '/'.implode('/', $parts);
+
+    }//end absolutePath()
+
+
+    /**
      * CodeSniffer alternative for realpath.
      *
      * Allows for PHAR support.
      *
-     * @param string $path The path to use.
+     * @param string  $path    The path to use.
+     * @param boolean $isStdin whether the file from STDIN.
      *
      * @return mixed
      */
-    public static function realpath($path)
+    public static function realpath($path, $isStdin=false)
```

Comment:
Optional helper parameter, consider to instead create a second function and call that second function when intended (functional decomposition instead of conditional code).